docs/programming_guide/execution_api_type/client_api.rst

```diff
@@ -159,24 +159,62 @@ If you are using PyTorch Lightning in your training code, you can check the
 Lightning API Module :mod:`nvflare.app_opt.lightning.api`.
 
 
+Client API communication patterns
+=================================
+
+.. image:: ../../resources/client_api.png
+    :height: 300px
+
+We offer various implementations of Client APIs tailored to different scenarios, each linked with distinct communication patterns.
+
+Broadly, we present in-process and sub-process executors. The in-process executor, slated for release in NVFlare 2.5.0,
+entails both training scripts and client executor operating within the same process. Communication between them occurs
+through an in-memory databus.
+
+On the other hand, the LauncherExecutor employs a sub-process to execute training scripts, leading to the client executor
+and training scripts residing in separate processes. Communication between them is facilitated by either CellPipe
+(default) or FilePipe.
+
+
+Choice of different Pipes
+=========================
+We suggest using the default setting with CellPipe for most users.
+
+CellPipe facilitates TCP-based cell-to-cell connections between the Executor and training script processes on
```

Comment:
I think this schema is also configurable?
It depends on the CellNet configuration of the whole system?